controllers/automation/policyautomation_controller.go

```diff
@@ -199,6 +201,121 @@ func (r *PolicyAutomationReconciler) Reconcile(
 			} else {
 				log.Info("All clusters are compliant. Doing nothing.")
 			}
+		} else if policyAutomation.Spec.Mode == "everyEvent" {
+			log := log.WithValues("mode", "everyEvent")
+			targetList := common.FindNonCompliantClustersForPolicy(policy)
+			eventTime := time.Now().UTC().Format(time.RFC3339)
+			if len(targetList) > 0 {
+				delayAfterRunSecondsStr := policyAutomation.Spec.DelayAfterRunSeconds
+				delayAfterRunSeconds, err := strconv.Atoi(delayAfterRunSecondsStr)
+				if err == nil {
+					log.Info("Can't read delayAfterRunSeconds, set to default value zero")
+					// if delayAfterRunSeconds is unset or isn't int, set to default value 0
+					delayAfterRunSeconds = 0
+				}
+				statusTemp := policyAutomation.Status
```

Comment:
This doesn't actually create a copy since `policyAutomation.Status` is a map. I also don't think you need to create a copy because `err := r.Get(ctx, request.NamespacedName, policyAutomation)` receives a copy from the cache.